### PR TITLE
Defining new contribution process

### PR DESCRIPTION
**Pull request by suava**
Android Starlark rules will soon be open sourced and released for customers to use and expand. It is important that we ensure that our user base is well represented and appropriately considered on the path to any contribution. This document serves as a guide for external parties to contribute to Android Starlark Rules productively and efficiently in partnership with the Mobile DevX Build team.

### CODE REVIEW

**Comment by suava** [comment]
I signed it!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,39 +1,78 @@
...
-even after your contribution becomes part of our codebase, so we need your
-permission to use and distribute your code. We also need to be sure of
-various other things — for instance that you'll tell us if you know that
-your code infringes on other people's patents. You don't have to sign
-the CLA until after you've submitted your code for review and a member has
-approved it, but you must do it before we can put your code into our codebase.
-
-### The small print
-Contributions made by corporations are covered by a different agreement than
+Want to contribute? Great! First, read this page. This document serves as a guide for you to contribute to Android Starlark Rules productively and efficiently in partnership with Google.
```
Comment (timpeut) [inline CONTRIBUTING.md]:
```suggestion
Want to contribute? Great! First, read this page. This document serves as a guide for you to contribute to the Android Starlark Rules productively and efficiently in partnership with Google.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,39 +1,78 @@
...
+
+1. Determine whether your contribution requires a lightweight change or a more complex change to the codebase.
+
+   1. Lightweight change: Simple bug fixes, small updates, internal cleanup, etc.
+   
+   1. Complex change: New features, large scale bug fixes, major refactoring, etc.
+
+1. If you have determined your change as lightweight, then you can skip to Step 7. You may be asked to follow the full process if your pull request reviewer believes a more thorough design is needed. Follow all steps if you have a more complex change.
+
+1. Author a brief contribution proposal document and submit to mobile-devx-build@google.com for review. Please use the following [template](https://docs.google.com/document/d/1QvyOCJGZivWFNm9VqH2Kp-vq_ELx0662vygVlbKOojM/edit?usp=sharing) for the document outlining:
```
Comment (timpeut) [inline CONTRIBUTING.md]: Template isn't accessible and emails bounce.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,39 +1,78 @@
...
+
+   1. Problem Statement
+   
+   1. Proposed Solution
+   
+   1. Alternative Considerations if any
+
+1. Out team will assign a reviewer and provide questions/comments on the design document.
```
Comment (timpeut) [inline CONTRIBUTING.md]:
```suggestion
1. Our team will assign a reviewer and provide questions/comments on the design document.
```

**Comment by jin** [comment]
As spoken offline, could you add something about the #android channel in Bazel's Slack? 

- https://bazelbuild.slack.com
- inviter: https://bazel-slackin.herokuapp.com

**Pull request merged**